src/carousel.js

```diff
@@ -57,9 +53,28 @@ class Carousel extends React.Component {
     this.refs.wrapper.removeEventListener('touchend', this.onTouchEnd, {capture: true})
   }
 
-  componentWillReceiveProps(nextProps) {
+  componentDidUpdate(_, prevState) {
+    if (this.state.frames.length && this.state.frames.length !== prevState.frames.length) {
+      // reset to default state
+      this.hideFrames()
+      translateXY(this.refs['f0'], 0, 0, 0)
```

Comment:
Should use `this.prepareAutoSlide()` instead of `translateXY(this.refs['f0'], 0, 0, 0)` to get all frames positioned correctly.

Reply:
@amio  just noticed the comments :/ will be able to get to it within 24 hours, thanks for the patience and I apologize for the latency.

Reply:
@amio this has been addressed